homeassistant/components/shelly/sensor.py

```diff
@@ -0,0 +1,80 @@
+"""Sensor for Shelly."""
+import aioshelly
+
+from homeassistant.components import sensor
+from homeassistant.const import TEMP_CELSIUS, TEMP_FAHRENHEIT
+from homeassistant.helpers.entity import Entity
+
+from . import ShellyBlockEntity, ShellyDeviceWrapper
+from .const import DOMAIN
+
+SENSORS = {
+    "extTemp": [None, sensor.DEVICE_CLASS_TEMPERATURE],
+    "humidity": ["%", sensor.DEVICE_CLASS_HUMIDITY],
+}
+
+
+async def async_setup_entry(hass, config_entry, async_add_entities):
+    """Set up sensors for device."""
+    wrapper = hass.data[DOMAIN][config_entry.entry_id]
+    sensors = []
+
+    for block in wrapper.device.blocks:
+        if block.type != "sensor":
+            continue
```

Comment:
This will block every sensors assigned to relays/devices, f.e. Shelly 2 `device` block contains energy and power sensors.
```
block: <device {'I': 4, 'D': 'device'}>
current values: {'cfgChanged': 0, 'power': 0.0, 'energy': 0, 'overpowerValue': 0.0, 'mode': 'relay'}
type: device
```

Reply:
Let's do that in the next PR. This PR was to get the foundation in place so it will be easier to add different values too :)